pxemgr/pxemanager_test.go

```diff
@@ -0,0 +1,156 @@
+package pxemgr
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/giantswarm/mayu-infopusher/machinedata"
+	"github.com/giantswarm/mayu/hostmgr"
+)
+
+const (
+	baseConfig = `default_coreos_version: myversion
+network:
+  ip_range:
+    start: 1.1.1.1
+    end: 1.1.1.2
+templates_env:
+  mayu_https_endpoint: https://mayu
+`
+	configOK             = baseConfig + "  storage: mystorage"
+	configErr            = baseConfig
+	lastStageCloudconfig = `before_key: ok
+{{if eq .TemplatesEnv.storage "mystorage"}}storage: mystorage{{end}}
+after_key: ok
+`
+)
+
+var (
+	dir      string
+	fakeEtcd *httptest.Server
+	pxeCfg   PXEManagerConfiguration
+	req      *http.Request
+	w        *httptest.ResponseRecorder
+	cluster  *hostmgr.Cluster
+)
+
+func setUp(t *testing.T) {
+	var err error
+	dir, err = ioutil.TempDir("", "pxmgr_cloudconfig_")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := ioutil.WriteFile(filepath.Join(dir, "config_ok.yaml"), []byte(configOK), 0644); err != nil {
```

Comment:
Just FYI, I think that using the real file system in tests is a good sign of code smell. This is not your fault in Mayu though. We tend to make external dependencies like the file system injectable and use abstracted libraries like https://godoc.org/github.com/spf13/afero.